pkg/asset/agent/manifests/agentclusterinstall.go

```diff
@@ -171,5 +181,83 @@ func (a *AgentClusterInstall) finish() error {
 		return errors.New("missing configuration or manifest file")
 	}
 
+	if err := a.validateIPAddressAndNetworkType().ToAggregate(); err != nil {
+		return errors.Wrapf(err, "invalid NetworkType configured")
+	}
+
 	return nil
 }
+
+// Sets the default network type to OVNKubernetes if it is unspecified in the
+// AgentClusterInstall or InstallConfig
+func setNetworkType(aci *hiveext.AgentClusterInstall, installConfig *types.InstallConfig,
+	warningMessage string) {
+
+	if aci.Spec.Networking.NetworkType != "" {
+		return
+	}
+
+	if installConfig != nil && installConfig.Networking != nil &&
+		installConfig.Networking.NetworkType != "" {
+		aci.Spec.Networking.NetworkType = installConfig.NetworkType
+		return
+	}
+
+	defaults.SetInstallConfigDefaults(installConfig)
```

Comment:
What's the reason to invoke such method here, since it seems that (potentially) could affect a completely different asset from the current one? If so, I'd find it confusing

Reply:
My understanding is that there is a [relationship between these two settings](https://github.com/openshift-metal3/dev-scripts/blob/master/agent/docs/config-presets-and-options.md#ip-stack). I assume it is because of technical limitations but I don't know the limitations of OpenShiftSDN and OVNKubernetes.

Reply:
My understanding is that there is a [relationship between these two settings](https://github.com/openshift-metal3/dev-scripts/blob/master/agent/docs/config-presets-and-options.md#ip-stack). I assume it is because of technical limitations but I don't know the limitations of OpenShiftSDN and OVNKubernetes.

Reply:
Okay I did some hunting in the OCP docs. TLDR: [OpenShiftSDN](https://docs.openshift.com/container-platform/4.11/networking/openshift_sdn/about-openshift-sdn.html) doesn't support IPv6 but [OVNKubernetes](https://docs.openshift.com/container-platform/4.11/networking/ovn_kubernetes_network_provider/about-ovn-kubernetes.html) does. Maybe some other things too.. but that is the big one that sticks out to me.

Reply:
In general... OVNKubernetes just has more features it seems. :) So if this is blank it would make sense to default to it.